Src/IronPython/Runtime/LiteralParser.cs

```diff
@@ -35,27 +35,41 @@ public static class LiteralParser {
             Debug.Assert(bytes != null);
             Debug.Assert(start + length <= bytes.Length);
 
-            string result = DoParseString(bytes, start, length, isRaw, isUniEscape: true, normalizeLineEndings: false, errorHandler);
+            string result = DoParseString(bytes.AsSpan(start, length), isRaw, isUniEscape: true, normalizeLineEndings: false, errorHandler);
 
             return result ?? bytes.MakeString(start, length);
         }
 
-        private static string DoParseString<T>(T[] data, int start, int length, bool isRaw, bool isUniEscape, bool normalizeLineEndings, ParseStringErrorHandler<T> errorHandler = default) where T : IConvertible {
-            Bytes bytesData = null;
+        internal static string ParseString(in ReadOnlySpan<byte> bytes, bool isRaw, ParseStringErrorHandler<byte> errorHandler) {
+            string result = DoParseString(bytes, isRaw, isUniEscape: true, normalizeLineEndings: false, errorHandler);
+
+            if (result == null) {
+                if (bytes.IsEmpty) {
```

Comment:
I agree. I had more improvements in mind here. It seems that we use the Latin-1 codec a lot, so I was thinking of making somewhere a global static property `Encoding Latin1` rather than calling `Encoding.GetEncoding` every time or looping ourselves. Just like [.NET has it](https://referencesource.microsoft.com/#mscorlib/system/text/encoding.cs,0f9d38a1189e6308) (and [.NET Core too](https://source.dot.net/#System.Private.CoreLib/Encoding.cs,0f9d38a1189e6308)), unfortunately, it is `private`.